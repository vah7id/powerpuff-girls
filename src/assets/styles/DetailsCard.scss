$breakpoint-tablet: 768px;

.DetailsCard {
  padding-top: 25px;
  display: flex;

  h2 {
    font-size: 2em;
    padding: 0 20px 0 0;
  }

  .col {
    flex: 1;

    p {
      padding: 0 20px 0 0;
      line-height: 24px;
    }
  }

  img {
    border: 1px solid #222;
    margin-top: 35px;
    flex: 1;
    max-width: 400px;

    @media (max-width: $breakpoint-tablet) {
      margin: 0 auto;
      width: 100%;
      display: block;
    }
  }

  @media (max-width: $breakpoint-tablet) {
    display: block;
  }
}
