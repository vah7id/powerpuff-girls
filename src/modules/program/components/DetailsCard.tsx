import * as React from "react";
import { Program } from "../interfaces";
import placeholderImage from "../../../assets/images/placeholder.png";
import '../../../assets/styles/DetailsCard.scss';

export const DetailsCard: React.FC<Program> = ({ name, image, summary }) => (
    <div className={'DetailsCard'}>
        <div className={'col'}>
            <h2>{name}</h2>
            <p dangerouslySetInnerHTML={{__html: summary}} />
        </div>
        <img src={image?.medium || placeholderImage} alt={name} />
    </div>
);

export default DetailsCard;
