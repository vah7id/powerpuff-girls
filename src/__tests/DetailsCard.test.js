import React from 'react';
import { shallow } from 'enzyme';
import DetailsCard from '../modules/program/components/DetailsCard';

// test cases
const propsMocks = [{
    name: 'Powerpuff Girls 1',
    summary: 'summary bla bla...',
    image: null
}, {
    name: 'Powerpuff Girls 2',
    summary: '<p>summary bla bla...</p>',
    image: {
        medium: 'http://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg',
        original: 'http://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg'
    }
}];

describe('DetailsCard component rendering', () => {

    it('should match the snapshot', () => {
        const container = shallow(<DetailsCard {...propsMocks[0]} />);
        expect(container.html()).toMatchSnapshot();
    });

    it('should render all the required fields with placeholder image', () => {
        const wrapper = shallow(<DetailsCard {...propsMocks[0]} />);
        expect(wrapper.find('h2').text()).toEqual(propsMocks[0].name);
        expect(wrapper.find('p').html()).toEqual(`<p>${propsMocks[0].summary}</p>`);
        expect(wrapper.find('img').props().src).toEqual('placeholder.png');
    });

    it('should render all the required fields with original image', () => {
        const wrapper = shallow(<DetailsCard {...propsMocks[1]} />);
        expect(wrapper.find('h2').text()).toEqual(propsMocks[1].name);
        expect(wrapper.find('p').html()).toEqual(`<p>${propsMocks[1].summary}</p>`);
        expect(wrapper.find('img').props().src).toEqual(propsMocks[1].image.medium);
    });
});
