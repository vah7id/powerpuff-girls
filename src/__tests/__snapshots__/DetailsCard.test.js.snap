// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsCard component rendering should match the snapshot 1`] = `"<div class=\\"DetailsCard\\"><div class=\\"col\\"><h2>Powerpuff Girls 1</h2><p>summary bla bla...</p></div><img src=\\"placeholder.png\\" alt=\\"Powerpuff Girls 1\\"/></div>"`;
